packages/state-manager/src/sagas/publicChannels/publicChannels.selectors.ts

```diff
@@ -200,35 +205,53 @@ export const dailyGroupedCurrentChannelMessages = createSelector(displayableCurr
   return result
 })
 
-export const currentChannelMessagesMergedBySender = createSelector(dailyGroupedCurrentChannelMessages, groups => {
-  const result: MessagesDailyGroups = {}
-  for (const day in groups) {
-    result[day] = groups[day].reduce((merged: DisplayableMessage[][], message: DisplayableMessage) => {
-      // Get last item from collected array for comparison
-      if (!merged.length) {
-        merged.push([message])
+/**
+ * Channel messages grouped by day and then additionally by sender (if
+ * there are successive messages by the same sender). Also adds a
+ * profile photo to the first message of each user.
+ */
+export const currentChannelMessagesMergedBySender = createSelector(
+  dailyGroupedCurrentChannelMessages,
+  userProfiles,
+  (groups: MessagesGroupsType, userProfiles: Record<string, UserProfile>) => {
+    const result: MessagesDailyGroups = {}
+    for (const day in groups) {
+      result[day] = groups[day].reduce((merged: DisplayableMessage[][], message: DisplayableMessage) => {
+        if (!merged.length) {
+          // Add profile photo to user's first message
+          if (message.pubKey in userProfiles) {
+            message.photo = userProfiles[message.pubKey].profile.photo
```

Comment:
The diff is large, but I really just added this line and a similar one below

Reply:
How about moving it to the `displayableCurrentChannelMessages` selector so that each message has a sender photo assigned? This way it wouldn't be necessary to duplicate this statement a few lines below